ygen/codegen_test.go

```diff
@@ -641,3 +624,96 @@ func TestFindRootEntries(t *testing.T) {
 		}
 	}
 }
+
+func TestGenerateProto3(t *testing.T) {
+	tests := []struct {
+		name           string
+		inFiles        []string
+		inIncludePaths []string
+		inConfig       GeneratorConfig
+		// wantOutputFiles is a map keyed on protobuf package name with a path
+		// to the file that is expected for each package.
+		wantOutputFiles map[string]string
+		wantErr         bool
+	}{{
+		name:     "simple protobuf test with compression",
+		inFiles:  []string{filepath.Join(TestRoot, "testdata", "proto", "proto-test-a.yang")},
```

Comment:
To avoid repeating this Join, you could define a subdir e.g. baseDir := TestRoot + "/testdata/proto" and use filepath.Join in the test loop to add just the filename. btw. filepath will correctly handle Join("a//b/", "/c") -> "a/b/c"

Reply:
Since we need a filepath.Join everywhere, it doesn't seem like it removes enough to do this. Prefer this approach.